marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings.tsx

```diff
@@ -271,22 +271,34 @@ const AdvancedSettings: React.FC<Props> = (props) => {
   };
 
   const getPayload = () => {
-    return {
+    let payload = 
+    {
       collections: defaultCollections,
       additionalCollections: additionalCollections,
-      sourceDatabase: usesSourceDatabase ? sourceDatabase: null,
       targetDatabase: targetDatabase,
       targetFormat: targetFormat,
       permissions: targetPermissions,
       headers: isEmptyString(headers) ? {} : parseJSON(headers),
       interceptors: isEmptyString(interceptors) ? [] : parseJSON(interceptors),
       provenanceGranularityLevel: provGranularity,
-      validateEntity: validateEntity,
       batchSize: batchSize,
       customHook: isEmptyString(customHook) ? {} : parseJSON(customHook),
-      targetCollections: usesAdvancedTargetCollections ? targetCollections : undefined,
-      additionalSettings: stepType === "custom" || isCustomIngestion ? parseJSON(additionalSettings) : {}
     };
+
+    if (usesSourceDatabase) {
```

Comment:
Is there any reason why ternary operators are replaced with "if" statements ? ternary operator are more terse.

Reply:
Ternary operators will leave behind fields that shouldn't be there.  For example,
`sourceDatabase: usesSourceDatabase ? sourceDatabase: null`
This will send the property `sourceDatabase: null` to the step as opposed to not having a sourceDatabase property at all.
To ensure that the step does not have a validateEntity prop at all, an if() statement is needed.